src/backend/distributed/deparser/qualify_text_search_stmts.c

```diff
@@ -92,7 +101,16 @@ QualifyDropTextSearchDictionaryStmt(Node *node)
 
 		if (!schemaName)
 		{
-			Oid tsdictOid = get_ts_dict_oid(objName, false);
+			Oid tsdictOid = get_ts_dict_oid(objName, true);
```

Comment:
I think deciding anything in a `Qualify*` function is really hard to follow, and probably not right. So, we return unqualified version of the input, which is NOT expected.  

One idea I'm tinkering is whether we should add `exists()` function to `DistributeObjectOps`. 

With that, we can do something like below in the pre-process:

```
if (ops && ops->exists)
{
    objectExists = ops->exists(parsetree)
}

if (objectExists)
{
   .. rest of the qualify, preprocess is called
}
```

With that, we'd avoid lots of code changes in this PR, and prevent any future such bugs. If the object does not exist, we just let Postgres do its job.

For now, the default/generic `exists` function can blindly return `true`, and for the objects that fails in check-vanilla (such as this one) we write proper `exists` function.

Does that make sense to you as well? If so, this should be done in a separate PR